xarray/core/dataset.py

```diff
@@ -9147,7 +9147,12 @@ def pad(
             int | tuple[int, int] | Mapping[Any, tuple[int, int]] | None
         ) = None,
         constant_values: (
-            float | tuple[float, float] | Mapping[Any, tuple[float, float]] | None
+            float
+            | tuple[float, float]
+            | Mapping[
+                Any, float | tuple[float, float] | Mapping[Any, tuple[float, float]]
+            ]
+            | None
```

Comment:
This is a pretty gnarly type signature. Same with the docstring. But I don't think there's much we can do to simplify it.

We possibly could make `float | tuple[float, float] | Mapping[Any, tuple[float, float]]` into an alias?

Reply:
Yeah that's true, I'll do that. Is there a convention for the location of type alias? or should I just define it before the function?

Reply:
So, I changed it slightly. I created these aliases:
```
T_PadConstantValues = float | tuple[float, float]
T_VarPadConstantValues = T_PadConstantValues | Mapping[Any, T_PadConstantValues]
T_DatasetPadConstantValues = T_VarPadConstantValues | Mapping[Any, T_VarPadConstantValues]
```
This way it is a bit more organized. Also changed the implementation of `pad` for variables so that it can make use of `Mapping[Any, float]` to keep consistency with the `pad` function for datasets.

What do you think?
(Also let me know if the location and import is correct, I don't fully understand when to import types under the `if TYPE_CHECKING:` block or above)

Reply:
Looks good, thanks!

(others may have refinements...)